show-device-location-using-indoor-positioning/src/main/java/com/esri/arcgismaps/sample/showdevicelocationusingindoorpositioning/MainActivity.kt

```diff
@@ -0,0 +1,365 @@
+            ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED ||
+            (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S &&
+                    ContextCompat.checkSelfPermission(this, Manifest.permission.BLUETOOTH_SCAN) != PackageManager.PERMISSION_GRANTED)
+        ) {
+            val requestPermissions = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+                arrayOf(
+                    Manifest.permission.ACCESS_FINE_LOCATION,
+                    Manifest.permission.ACCESS_COARSE_LOCATION,
+                    Manifest.permission.BLUETOOTH_SCAN
+                )
+            } else {
+                arrayOf(
+                    Manifest.permission.ACCESS_FINE_LOCATION,
+                    Manifest.permission.ACCESS_COARSE_LOCATION
+                )
+            }
+            ActivityCompat.requestPermissions(this, requestPermissions, requestCode)
+        } else {
+            // permission already given, so no need to request
+            setUpMap()
+        }
+    }
+
+    /**
+     * Set up the connection between the device and the portal item
+     * and set the [mapView] using the [portalItem]
+     */
+    private fun setUpMap() {
+        // load the portal and create a map from the portal item
+        val portalItem = PortalItem(
+            Portal("https://www.arcgis.com/"),
+            "8fa941613b4b4b2b8a34ad4cdc3e4bba"
+        )
+        val map = ArcGISMap(portalItem)
+        mapView.map = map
+        lifecycleScope.launch {
+            map.load().onSuccess {
+                val featureTables = map.tables
+                // check if the portalItem contains featureTables
+                if (featureTables.isNotEmpty()) {
+                    setUpLoadTables(featureTables)
+                } else {
+                    showError("Map does not contain feature tables")
+                }
+            }.onFailure {
+                // if map load failed, show the error
+                showError("Error Loading Map")
+            }
+        }
+    }
+
+    /**
+     * Load each feature table and setup the IndoorsLocationDataSource for each table loaded
+     */
+    private fun setUpLoadTables(featureTables: MutableList<FeatureTable>) {
+        lifecycleScope.launch {
+            featureTables.forEach { featureTable ->
+                // load each FeatureTable
+                featureTable.load().onSuccess {
+                    setupIndoorsLocationDataSource(featureTable)
+                }.onFailure {
+                    showError("Error loading FeatureTable: ${it.message}")
+                }
+            }
+        }
+    }
+
+    /**
+     * Sets up the [mIndoorsLocationDataSource] using the IPS_Positioning table
+     */
+    private fun setupIndoorsLocationDataSource(featureTable: FeatureTable) {
+        // positioningTable needs to be present
+        if (featureTable.tableName == "IPS_Positioning") {
```

Comment:
Can this check be done in the caller function? 
```kotlin
if (featureTable.tableName == "IPS_Positioning") {
    setupIndoorsLocationDataSource(featureTable)
}
```

Reply:
I modified it as below
if (featureTable.tableName != "IPS_Positioning") {
            showError("Positioning Table not found in FeatureTables")
            return
        } // IPS_Positioning table needs to be present
        else if (featureTable.tableName == "IPS_Positioning") {
        ....
        }
        
I think either approach should be fine.

Reply:
The `else` condition is not nesseary here since if `featureTable.tableName != "IPS_Positioning"` is true the function will return.

Reply:
Done! :)

Reply:
```
// IPS_Positioning table needs to be present
                    if (featureTable.tableName == "IPS_Positioning") {
                        setupIndoorsLocationDataSource(featureTable)
                    } else {
                        showError("Positioning Table not found in FeatureTables")
                    }
```